app/services/notify_service.rb

```diff
@@ -16,6 +17,14 @@ def call(recipient, activity)
 
   private
 
+  def visibility
+    unless !@visibility.nil? || @activity.is_a?(Follow) || @activity.is_a?(FollowRequest)
+      status = @activity.is_a?(Status) ? @activity : @activity.status
```

Comment:
This also blocks favorite notification for DMs you sent, not only mention notification for received DMs. It seems unwanted behavior for this feature.

Reply:
I don't agree that it's unwanted behavior - it stops someone from spamming the favorite button to flood your notifications.

Reply:
Hmm, okay. although `Block only Direct Messages from people you don't follow` sounded that it only blocks DM itself (i.e. mentions).